[DSCP-471] More specific `Arbitrary` instance for `CertificateMeta`

flyingleafe opened the pull request:
### Description

Changes `Arbitrary` instance so that dates make sense and text fields do not contain incomprehensible characters. Also, removes `DefPageSize` from pagination parameters.

### YT issue

https://issues.serokell.io/issue/DSCP-471

### Checklist

Hint: a perfect PR has all the checkmarks set.

Possible related changes (conditional):
- [x] I added tests if required, in case they are:
  - Covering new introduced functionality (feature).
  - Regression tests preventing the bug from silently reappearing again (bugfix).
- [x] I have checked the [sample config](../tree/master/docs/config-full-sample.yaml) and [launch scripts](../tree/master/scripts/launch) and updated them if it was necessary (especially if executables or CLIs were changed).
- [x] I have checked READMEs and changed them accordingly if it's necessary (the main [README.md](../tree/master/README.md) as well as subproject READMEs for all related executables).
- [x] If a public API structure or/and data serialization was changed, I made sure that these changes are reflected in the:
  - [Swagger](../tree/master/specs/disciplina) API specs.
  - Any [related documentation](../tree/master/docs/api-types.md).
- [x] If any public documentation was written, I have spellchecked it.

Stylistic (obligatory):
- [x] My commit history is clean, descriptive and do not contain merge or revert commits or commits like "WIP".
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).
- [x] My code is documented (haddock) and these docs are decent (full enough and spellchecked).

flyingleafe committed:
[DSCP-471] More specific `Arbitrary` instance for `CertificateMeta`

Martoon-00 commented on core/src/Dscp/Core/Arbitrary.hs:
```diff
@@ -246,7 +265,8 @@ instance ArbitraryMixture BlockBody
 
 instance Arbitrary ItemDesc where
-    arbitrary = arbitrary `suchThatMap` (rightToMaybe . toItemDesc)
+    arbitrary = (toBase64 @ByteString <$> arbitrary)
```
I don't like it, tests should cover any possible scenarious and it would be better to create a separate generator for such smart thing.
I cannot image a case when this change causes us to miss a bug though.

flyingleafe replied:
Well, type `ItemDesc` assumes that its contents should be somewhat readable, and set of `base64` encoded strings is still quite wide. 
Basically, this change might cause us to miss a bug if we processed the contents of `ItemDesc`s in some way which is more complex than taking a hash of it and putting it into the database, which we do not and I cannot imagine a plausible situation when we will.

Martoon-00 commented on core/src/Dscp/Core/Arbitrary.hs:
```diff
@@ -118,8 +120,25 @@ instance Arbitrary EducationForm where
+        -- actual diploma numbers
+        cmNumber <- arbitrary `suchThat` (> 1000000)
+        -- Document should be issued in the year when studies are ended
+        cmIssueDate <- arbitrary `suchThat` (\issueDate -> getYear issueDate == cmEndYear)
```
Kek, requests to the endpoint hang. And something in this line makes me guess a suspect.

flyingleafe replied:
@Martoon-00 resolved. But the removal of `DefPageSize` required some fixes in `servant-util`, check out: https://github.com/serokell/servant-util/pull/3

flyingleafe committed:
[DSCP-471] Fix bugs in random generation of `CertificateMeta`

Also, the `DefPageSize` for `GetCertificates` is enlarged - this is
a temporary hack to reduce the confusion of frontend engineers before
`DefPageSize` is able to be removed.

flyingleafe committed:
[DSCP-471] Remove Educator name from `CertificateMeta`

flyingleafe committed:
Add LaTeX dependencies to `shell-native.nix`

flyingleafe committed:
[DSCP-471] Use lazy bytestrings for PDF generation

flyingleafe committed:
[DSCP-471] Add `EducatorInfo` datatype to latex generator.

flyingleafe committed:
[DSCP-471] Fix model spec for `CertificateMeta`

Martoon-00 commented on pdfs/src/Pdf/FromLatex.hs:
```diff
@@ -35,20 +37,27 @@ import Dscp.Core.Foundation.Educator
-    Text.toStrict $ Text.toLazyText $ make (lang, cert)
+-- | Datatype containing information about Educator which is
+-- required in order to render a certificate.
+data EducatorInfo = EducatorInfo
```
Oh kek, we already have one `EducatorInfo` [here](https://github.com/DisciplinaOU/disciplina/blob/martoon%2Fdscp457-bump/educator/src/Dscp/Educator/Web/Educator/Types.hs#L80).
Could you rename that to `EducatorStatusInfo` (to comply with endpoint name)?

flyingleafe replied:
Which endpoint name, exactly?....
Better to make it `CertificateIssuerInfo` then, frankly.

Martoon-00 replied:
`api/educator/v1/status` endpoint, but do as you see fit.

Martoon-00 commented on shell-native.nix:
```diff
@@ -18,7 +18,30 @@ stdenv.mkDerivation rec {
+        xcolor
+        pgf
+        fontspec;
+    })
```
Life is harsh

Martoon-00 reviewed (approved):
Welcome back to programming )

flyingleafe committed:
[DSCP-471] Rename `EducatorInfo` in a less confusing way.
